evap/results/views.py

```diff
@@ -86,10 +86,7 @@ def update_evaluation_template_cache(evaluations):
 
 def update_template_cache_of_published_evaluations_in_course(course):
     course_evaluations = course.evaluations.filter(state="published")
-    for course_evaluation in course_evaluations:
-        _delete_evaluation_template_cache_impl(course_evaluation)
-    _delete_course_template_cache_impl(course)
-    warm_up_template_cache(course_evaluations)
+    update_template_cache(course_evaluations)
```

Comment:
This is why the test fails - Sorry, I messed this up because we run into a strange scenario here: If the course does not have evaluations anymore, we still need to do `_delete_course_template_cache_impl` - otherwise, course_evaluations will be empty -> the cache for the course will not be cleared. You could do:
```
if course_evaluations:
    update_template_cache(course_evaluations)
else:
    _delete_course_template_cache_impl(course)
    warm_up_template_cache(???) # or something else - do we need some cache here? Or will a course without evaluations always be filtered from results? This is a struggle with the old as well as the new version of this method.
```
_if_ we want it to be possible to have a course without evaluations. In this case, we'd expect two calls on `_delete_course_template_cache_impl` and 0 calls on `warm_up_template_cache`, since the evalulation is not published and therefore filtered. I get that with the modifications. It may be easier to just revert these three lines if we plan on changing the caching anyway (only cache per course)